api/capi/src/nnstreamer-capi-single.c

```diff
@@ -374,129 +374,173 @@ ml_single_set_info_in_handle (ml_single_h single, gboolean is_input,
 }
 
 /**
- * @brief Opens an ML model and returns the instance as a handle.
+ * @brief Internal function to create and initialize the single handle.
+ */
+static ml_single *
+ml_single_create_handle (ml_nnfw_type_e nnfw)
+{
+  ml_single *single_h;
+  GError *error;
+
+  single_h = g_new0 (ml_single, 1);
+  if (single_h == NULL) {
+    ml_loge ("Failed to allocate the single handle.");
+    return NULL;
+  }
+
+  single_h->filter = g_object_new (G_TYPE_TENSOR_FILTER_SINGLE, NULL);
+  if (single_h->filter == NULL) {
+    ml_loge ("Failed to create a new instance for filter.");
+    g_free (single_h);
+    return NULL;
+  }
+
+  single_h->magic = ML_SINGLE_MAGIC;
+  single_h->timeout = SINGLE_DEFAULT_TIMEOUT;
+  single_h->nnfw = nnfw;
+  single_h->state = IDLE;
+  single_h->ignore_output = FALSE;
+
+  ml_tensors_info_initialize (&single_h->in_info);
+  ml_tensors_info_initialize (&single_h->out_info);
+  g_mutex_init (&single_h->mutex);
+  g_cond_init (&single_h->cond);
+
+  single_h->thread = g_thread_try_new (NULL, invoke_thread, (gpointer) single_h, &error);
+  if (single_h->thread == NULL) {
+    ml_loge ("Failed to create the invoke thread, error: %s.", error->message);
+    g_clear_error (&error);
+    ml_single_close (single_h);
```

Comment:
Q. `g_thread_join (NULL)` (called in `ml_single_close()`) is safe?

Reply:
g_thread_join() internally checks null ptr.
```
g_thread_join (GThread *thread)
{
  ...
  g_return_val_if_fail (thread, NULL);
```